pkg/utils/utils.go

```diff
@@ -253,7 +253,50 @@ func ReadScratchNetConf(cRefPath string) ([]byte, error) {
 // CleanCachedNetConf removed cached NetConf from disk
 func CleanCachedNetConf(cRefPath string) error {
 	if err := os.Remove(cRefPath); err != nil {
-		return fmt.Errorf("error removing NetConf file %s: %q", cRefPath, err)
+		return fmt.Errorf("error removing NetConf file %s: %v", cRefPath, err)
 	}
 	return nil
 }
+
+func SaveAllocatedPCI(pciAddress, ns, dataDir string) error {
```

Comment:
Shall we wrap these three methods in an interface ? 

i know cache is not implemented this way, but i do not think we need to take the same approach here.

it would be great if this would be its own package with interface definition, impl and UT.
or if you prefer, part of util package but a separate file e.g `pci_allocator.go` WDYT ?

the dataDir (currently `config.DefaultCNIDir`) should be provided when creating the interface implementation (i.e provided in the "New" function)

WDYT ?

later on we can move the caching methods to their own interface as well.